src/ansi-c/type2name.cpp

```diff
@@ -277,27 +279,52 @@ std::string type2name(const typet &type, const namespacet &ns)
 /// If we want utilities like dump_c to work properly identifiers
 /// should ideally always be valid C identifiers
 /// This replaces some invalid characters that can appear in type2name output.
-std::string type2identifier(const typet &type, const namespacet &ns)
+std::string type_name2type_identifier(const std::string &name)
```

Comment:
I still don’t like the `2` naming convention here, but changing that isn’t really in scope for this PR.